File: scripts/remove_na_riders.py
import os
from supabase import create_client
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Initialize Supabase client
supabase = create_client(
    os.getenv('VITE_SUPABASE_URL'),
    os.getenv('VITE_SUPABASE_ANON_KEY')
)

def remove_na_riders():
    """Remove riders with N/A deployment dates from Supabase"""
    
    print("🔍 Finding riders with N/A deployment dates...")
    
    # First, identify riders with N/A deployment dates
    response = supabase.table('riders').select('*').or('deployment_date.eq.N/A,deployment_date.is.null').execute()
    
    if not response.data:
        print("✅ No riders with N/A deployment dates found!")
        return
    
    na_riders = response.data
    print(f"📊 Found {len(na_riders)} riders with N/A deployment dates:")
    
    # Display riders to be deleted
    for rider in na_riders:
        print(f"  - ID: {rider.get('rider_id', 'N/A')}, Name: {rider.get('rider_name', 'N/A')}, Hub: {rider.get('operator_hub', 'N/A')}")
    
    # Confirm deletion
    confirm = input(f"\n⚠️  Are you sure you want to delete {len(na_riders)} riders? (yes/no): ")
    
    if confirm.lower() != 'yes':
        print("❌ Deletion cancelled.")
        return
    
    print("\n🗑️  Deleting riders...")
    
    # Delete riders with N/A deployment dates
    delete_response = supabase.table('riders').delete().or('deployment_date.eq.N/A,deployment_date.is.null').execute()
    
    if delete_response.data:
        print(f"✅ Successfully deleted {len(delete_response.data)} riders")
    else:
        print(f"❌ Error deleting riders: {delete_response.error}")
    
    # Verify cleanup
    print("\n🔍 Verifying cleanup...")
    total_response = supabase.table('riders').select('rider_id', count='exact').execute()
    remaining_na_response = supabase.table('riders').select('rider_id', count='exact').or('deployment_date.eq.N/A,deployment_date.is.null').execute()
    
    total_riders = total_response.count or 0
    remaining_na = remaining_na_response.count or 0
    
    print(f"📈 Total riders remaining: {total_riders}")
    print(f"📈 Riders with N/A deployment dates remaining: {remaining_na}")
    
    if remaining_na == 0:
        print("🎉 Cleanup completed successfully!")
    else:
        print("⚠️  Some riders with N/A deployment dates still remain.")

if __name__ == "__main__":
    remove_na_riders()

File: remove_na_riders.js
// Node.js script to remove riders with N/A deployment dates
// Run with: node remove_na_riders.js

import { createClient } from '@supabase/supabase-js'
import dotenv from 'dotenv'

// Load environment variables
dotenv.config()

const supabase = createClient(
  process.env.VITE_SUPABASE_URL,
  process.env.VITE_SUPABASE_ANON_KEY
)

async function removeNaRiders() {
  try {
    console.log('🔍 Finding riders with N/A deployment dates...')
    
    // First, identify riders with N/A deployment dates
    const { data: naRiders, error: fetchError } = await supabase
      .from('riders')
      .select('*')
      .or('deployment_date.eq.N/A,deployment_date.is.null')
    
    if (fetchError) {
      console.error('❌ Error fetching riders:', fetchError)
      return
    }
    
    if (!naRiders || naRiders.length === 0) {
      console.log('✅ No riders with N/A deployment dates found!')
      return
    }
    
    console.log(`📊 Found ${naRiders.length} riders with N/A deployment dates:`)
    
    // Display riders to be deleted
    naRiders.forEach(rider => {
      console.log(`  - ID: ${rider.rider_id || 'N/A'}, Name: ${rider.rider_name || 'N/A'}, Hub: ${rider.operator_hub || 'N/A'}`)
    })
    
    // Confirm deletion
    const readline = require('readline')
    const rl = readline.createInterface({
      input: process.stdin,
      output: process.stdout
    })
    
    const answer = await new Promise(resolve => {
      rl.question(`\n⚠️  Are you sure you want to delete ${naRiders.length} riders? (yes/no): `, resolve)
    })
    
    rl.close()
    
    if (answer.toLowerCase() !== 'yes') {
      console.log('❌ Deletion cancelled.')
      return
    }
    
    console.log('\n🗑️  Deleting riders...')
    
    // Delete riders with N/A deployment dates
    const { data: deletedRiders, error: deleteError } = await supabase
      .from('riders')
      .delete()
      .or('deployment_date.eq.N/A,deployment_date.is.null')
      .select()
    
    if (deleteError) {
      console.error('❌ Error deleting riders:', deleteError)
      return
    }
    
    console.log(`✅ Successfully deleted ${deletedRiders.length} riders`)
    
    // Verify cleanup
    console.log('\n🔍 Verifying cleanup...')
    
    const { count: totalRiders } = await supabase
      .from('riders')
      .select('*', { count: 'exact', head: true })
    
    const { count: remainingNa } = await supabase
      .from('riders')
      .select('*', { count: 'exact', head: true })
      .or('deployment_date.eq.N/A,deployment_date.is.null')
    
    console.log(`📈 Total riders remaining: ${totalRiders || 0}`)
    console.log(`📈 Riders with N/A deployment dates remaining: ${remainingNa || 0}`)
    
    if ((remainingNa || 0) === 0) {
      console.log('🎉 Cleanup completed successfully!')
    } else {
      console.log('⚠️  Some riders with N/A deployment dates still remain.')
    }
    
  } catch (error) {
    console.error('❌ Unexpected error:', error)
  }
}

removeNaRiders()
